src/mnt_setup_fnc.cpp

```diff
@@ -8,59 +8,88 @@
-	std::string sqrts,
+	std::string const DATADIR,
+	sqrts,
 	stripping,
 	year,
 	polarity,
 	stream;
 };
 
-void make_histogram(struct path_data path_in, std::string hist_name, std::string hist_setup, int const nbins, float const xmin, float const xmax, std::string expression, THStack *hist_stack) {
+struct input {
+	std::string const expression;
+	int const nbins;
+	float const xmin,
+	xmax;
+	std::string const label,
+	unit;
+};
+
+TH1F* make_TH1F(struct path_data path_in, struct input hist_input, std::string hist_name, std::string hist_setup) {
 	TChain ch("Z/DecayTree");
 	std::string path = path_in.DATADIR + path_in.sqrts + "TeV_" + path_in.year + "_" + path_in.stripping + "_" + path_in.polarity + "_" + path_in.stream + ".root";
 	ch.Add(path.c_str());
 	
-	TH1F *hist = new TH1F(hist_name.c_str(), hist_setup.c_str(), nbins, xmin, xmax);
-	expression += hist->GetName();
+	TH1F *hist = new TH1F(hist_name.c_str(), hist_setup.c_str(), hist_input.nbins, hist_input.xmin, hist_input.xmax);
+	std::string expression = hist_input.expression +">>"+ hist->GetName();
 	ch.Draw(expression.c_str());
-	hist_stack->Add(hist);
+	return hist;
 }
 
+void plot_histogram(struct path_data mnt_in, struct path_data sim_in, struct input hist_input) {
+	std::string hist_title = "Measurement vs Simulation for " + hist_input.label,
+		x_axis = hist_input.label + " " + hist_input.unit,
+		hist_text = hist_title + ";" + x_axis;
+	
+	TH1F *hist_mnt = make_TH1F(mnt_in, hist_input, "measurement", hist_text);
+	TH1F *hist_sim = make_TH1F(sim_in, hist_input, "simulation", hist_text);
+	
+	hist_sim->Scale(hist_mnt->Integral()/hist_sim->Integral());
+	
+	TCanvas canv;
+	hist_sim->SetStats(false);
+	hist_sim->Draw("HIST");
+	hist_mnt->Draw("SAME E");
+	std::string const filename = "Measurement_" + hist_input.label + ".png";
+	canv.SaveAs(filename.c_str());
+}
 
 int main() {
 
 	/*Measurement Chain*/
-	path_data measurement_in = {DATADIR, "5", "32", "2017", "Down", "EW"};
+	path_data const measurement_in = {DATADIR, "5", "32", "2017", "Down", "EW"};
 
 	/*Simulation Chain*/
-	path_data simulation_in = {DATADIR, "13", "28r1", "2016", "Down", "Z_Sim09h"};
+	path_data const simulation_in = {DATADIR, "13", "28r1", "2016", "Down", "Z_Sim09h"};
 	
-	/*Branch Input*/
-	std::string const muon_prop = "mum_PHI",
-		muon_prop_unit = " ";
 	
-	/*Histogram Set-up*/
-	std::string hist_title = "Measurement vs Simulation for " + muon_prop,
-		x_axis = muon_prop + muon_prop_unit,
-		hist_setup = hist_title + ";" + x_axis,
-		expression = muon_prop + ">>";
-		/*expression = "1.e-3*" + muon_prop + ">>"; version for PT in GeV*/
+	/*input property_in = {"expression", nbins, xmin., xmax., "label", "unit"};
+	plot_histogram(measurement_in, simulation_in, property_in);*/
 	
-	int const nbins = 100;
-	float const xmin = -4.,
-		xmax = 4.;
+	/*mup*/
+	input const mup_PT_in = {"1.e-3*mup_PT", 100, 15., 60., "mup_PT", "(GeV)"};
+	plot_histogram(measurement_in, simulation_in, mup_PT_in);

```

Comment:
You could write this more concisely and expressing it as "do something for this list of plot configurations".
```cpp
std::vector<input> plot_configurations;
plot_configurations.emplace_back("1.e-3*mup_PT", 100, 15., 60., "mup_PT", "(GeV)");
\\etc...
for (auto const & plot_configuration : plot_configurations ){
     plot_histogram(measurement_in,simulation_in,plot_configuration);
}
```
```

Reply:
P.s. some of what I write is C++11 standard. I'm guessing that you will get at least a C++11 compatible `gcc` version. If you get any errors about e.g. not knowing what `auto` means then you can add `std=c++11` or `std=c++14`... to the build command in the `Makefile`.